lib/tapioca/gem/listeners/source_location.rb

```diff
@@ -52,7 +52,7 @@ def add_source_location_comment(node, file, line)
           realpath = path.realpath.to_s
 
           path = if realpath.start_with?(gem.full_gem_path)
-            "#{gem.name}-#{gem.version}/#{path.realpath.relative_path_from(gem.full_gem_path)}"
+            "#{gem.name}-VERSION/#{path.realpath.relative_path_from(gem.full_gem_path)}"
```

Comment:
I love the idea. Do we then need another scheme for Ruby lib paths?

Reply:
> Do we then need another scheme for Ruby lib paths?
I think not. IMO, we are confounding where the gem is installed with where inside the gem this code is coming from. We want to encode the latter and leave it up to the client to resolve where the specified gem/version should be found.

For example, it could be the case that on my system I have the gem `foo@2.0.0` installed in the Ruby lib path, but on a colleague's machine `foo@2.0.0` is installed by Bundler/Rubygems to the gem path (imagine the colleague is using an older Ruby that bundles with `foo@1.0.0` and `foo@2.0.0` was a dependency in the lockfile). 

It should be the responsibility of the client to resolve `foo@1.0.0` to a path on disk and to then resolve the `path` relative to that location.

Reply:
Sorry, I wasn't clear. I'm not talking about where the gem is installed.

I mean when methods are defined not inside a gem, but inside the Ruby standard library. Like in this [test](https://github.com/Shopify/tapioca/blob/a51936a23a3fc947b2de93c8873022bece2106ec/spec/tapioca/gem/pipeline_spec.rb#L4022), where `Mutex_m` method locations are in the Ruby path - not inside any gems.

Those files are inside `RbConfig::CONFIG["rubylibdir"]` and not `Bundler.bundle_path`, so we need some way of differentiating between gems and the stdlib.

Reply:
So, is your suggestion to do `source://mutex.rb#12` and let the client figure out where it is?

Reply:
More like `source://mutex_m@0.1.1/mutex_m.rb#12`, but yeah. Since, like I said in my comment above, on another machine the `0.1.1` version of the `mutex_m` gem can be installed in the normal gem path.